### PR TITLE
feat: space3 genesis - open-edition contract

### PR DESCRIPTION
**Pull request by thanhle-ancient8**

### CODE REVIEW

File: contracts/open-edition/Space3Genesis.sol [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        return _nextTokenId;
+    }
+
+    modifier onlyInPublicMint() {
+        require(
+            block.timestamp >= startTime && block.timestamp < endTime,
+            "Not in public mint period"
+        );
+        _;
+    }
```
Comment (sangtran-a8) [inline contracts/open-edition/Space3Genesis.sol:37]: pls re-organize modifiers

File: contracts/open-edition/Space3Genesis.sol [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        endTime = endTime_;
+    }
+
+    function nextTokenId() public view returns (uint256) {
+        return _nextTokenId;
+    }
+
+    modifier onlyInPublicMint() {
+        require(
+            block.timestamp >= startTime && block.timestamp < endTime,
```
Comment (sangtran-a8) [inline contracts/open-edition/Space3Genesis.sol:33]: `block.timestamp < endTime` or `block.timestamp <= endTime`

File: contracts/open-edition/Space3Genesis.sol [hunk 3]
```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+
+contract Space3Genesis is ERC721Enumerable {
+    uint256 private _nextTokenId;
+    string private _uri;
+    uint256 public immutable startTime;
```
Comment (sangtran-a8) [inline contracts/open-edition/Space3Genesis.sol:9]: why `immutable`??? just onetime???

File: contracts/open-edition/Space3Genesis.sol [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    uint256 public immutable endTime;
+
+    modifier validPeriod(uint256 startTime_, uint256 endTime_) {
```
Comment (sangtran-a8) [inline contracts/open-edition/Space3Genesis.sol:12]: I think we don't need suffix _ here bro (optional)

File: contracts/open-edition/Space3Genesis.sol (hunk 2 above)
Comment (thanhle-ancient8) [reply contracts/open-edition/Space3Genesis.sol:33]: OR??

File: contracts/open-edition/Space3Genesis.sol (hunk 3 above)
Comment (thanhle-ancient8) [reply contracts/open-edition/Space3Genesis.sol:9]: yes, confirmed

File: contracts/open-edition/Space3Genesis.sol (hunk 4 above)
Comment (thanhle-ancient8) [reply contracts/open-edition/Space3Genesis.sol:12]: If you don't get it, try it yourself, the `startTime` has already been declared in line 9

File: contracts/open-edition/Space3Genesis.sol (hunk 2 above)
Comment (thanhle-ancient8) [reply contracts/open-edition/Space3Genesis.sol:33]: Gotcha, => `endTime` is not included, use `<`

**Review by robertnguyen98 — APPROVED** [review]

**Pull request merged**